lib/led-matrix/include/LEDMatrix.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include "hw_bridge.h"
+
+// Flags
+#define START_FLASH 1UL  // 01
+#define ENDED_FLASH 2UL  // 10
+
+class LEDMatrix {
+ private:
+  DigitalOut m_RChannel;
+  DigitalOut m_GChannel;
+  DigitalOut m_BChannel;
+
+  volatile bool m_flashing_red, m_flashing_green, m_flashing_blue;
+  volatile bool m_continue_flashing;
+
+  EventFlags m_event_flags;
+  Thread m_lightsThread;
+  static constexpr auto PERIOD_DELAY = 500ms;
+
+  // Thread to take care of flashing.
+  void flashing();
+
+  // Turn off all the LEDs. Call setColor(0,0,0).
+  void setColor(bool R, bool G, bool B);
+
+ public:
+  // Define matrix by the pins it is connected to.
+  LEDMatrix(PinName R, PinName G, PinName B);
+
+  // Clear lights.
+  ~LEDMatrix();
+
+  // Set the state of the LEDs.
+  void setState(HWBRIDGE::LEDMATRIX::LEDMatrixState state);
+
+  // Start flashing() thread.
+  void setFlashColor(bool R, bool G, bool B);
+
+  // Stop flashing() thread and call setColor().
+  void setSolidColor(bool R, bool G, bool B);
```

Comment:
Make sure these comments describe the outcome of these functions. The user of this driver doesn't necessarily care that you're starting the flashing thread or calling setColor() internally, they care about the fact that setFlashColor() makes the led matrix periodically flash a certain color, and that setSolidColor() sets the led matrix to a solid color given by the RGB values.

Reply:
That's a good, point, thanks!